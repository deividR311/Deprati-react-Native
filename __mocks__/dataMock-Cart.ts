export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
