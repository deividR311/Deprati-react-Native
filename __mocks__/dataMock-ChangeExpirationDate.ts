export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
