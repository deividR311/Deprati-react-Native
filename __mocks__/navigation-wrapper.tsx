import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
