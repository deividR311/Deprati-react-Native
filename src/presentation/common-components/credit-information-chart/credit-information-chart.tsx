import React from 'react';
import { Text, View } from 'react-native';
import { CreditInformationChartProps } from './interface';
import { Styles } from './credit-information-chart.stylesheet';
import { DonutChart } from './donut-chart';
import { FontStyles } from '../../../application/common/fonts';
import { capitalize } from '../../../application/utils';

export const CreditInformationChart: React.FC<
  CreditInformationChartProps
> = props => {
  const {
    constentChart,
    ownerFullname = '',
    numeroTarjetaAdicionalDisplay = '',
    affiliateDate = '',
    totalAmount = 1,
    amountSpent = 0,
    availableAmount = 0,
    showAffiliateDate = true,
    showDisplayCard = true,
    displayCartType = 'small',
    invertValues = false,
    wrapperChartInCard = false
  } = props;

  return (
    <View style={Styles.container}>
      <View
        style={[
          Styles.textContainer,
          displayCartType === 'large' ? Styles.largeContainerCard : undefined
        ]}>
        {showDisplayCard && displayCartType === 'small' && (
          <View style={Styles.leftTextContainer}>
            <Text style={[FontStyles.Caption]}>{ownerFullname}</Text>
            <Text style={[FontStyles.Caption]}>
              {numeroTarjetaAdicionalDisplay}
            </Text>
          </View>
        )}
        {showAffiliateDate && displayCartType === 'small' && (
          <Text style={[FontStyles.Caption, FontStyles.MutedColor]}>
            Cliente desde {affiliateDate}
          </Text>
        )}
        {showDisplayCard && displayCartType === 'large' && (
          <View style={[Styles.leftTextContainerLarge]}>
            <Text style={Styles.largeTitle}>Datos del cliente</Text>
            <View style={Styles.tableRow}>
              <Text style={[Styles.cellCaption]}>Nombre:</Text>
              <Text numberOfLines={1} style={[Styles.cellValue]}>
                {ownerFullname
                  ?.split(' ')
                  .map(n => capitalize(n, false))
                  .join(' ')}
              </Text>
            </View>
            <View style={Styles.tableRow}>
              <Text style={[Styles.cellCaption]}>CÃ³digo del cliente:</Text>
              <Text style={[Styles.cellValue]}>
                {numeroTarjetaAdicionalDisplay}
              </Text>
            </View>
            <View style={Styles.tableRow}>
              <Text style={[Styles.cellCaption]}>Cliente desde:</Text>
              <Text style={[Styles.cellValue]}>{affiliateDate}</Text>
            </View>
          </View>
        )}
      </View>
      <View
        style={[
          Styles.chartContainer,
          wrapperChartInCard ? Styles.wrapperChartInCard : undefined
        ]}>
        <View style={Styles.graphicContainer}>
          <DonutChart
            invertValues={invertValues}
            radius={107}
            maxIndicatorValue={totalAmount}
            indicatorValue={amountSpent}
          />
        </View>
        <View>
          {!invertValues && (
            <>
              <Text
                style={[
                  FontStyles.Body_2,
                  FontStyles.MutedColor,
                  FontStyles.Center
                ]}>
                Cupo utilizado
              </Text>
              <Text
                style={[
                  FontStyles.H1_Headline,
                  FontStyles.PrimaryColor,
                  { marginBottom: 8 }
                ]}>
                ${constentChart?.amountSpent ?? '0,00'}
              </Text>
            </>
          )}

          <Text
            style={[
              FontStyles.Body_2,
              invertValues ? FontStyles.Bold : undefined,
              invertValues ? FontStyles.DarkColor : FontStyles.MutedColor,
              FontStyles.Center,
              invertValues ? { marginTop: 8 } : undefined
            ]}>
            Cupo disponible
          </Text>
          <Text
            style={[
              invertValues ? FontStyles.H1_Headline : FontStyles.Body_2,
              FontStyles.Center,
              { marginBottom: 6 }
            ]}>
            ${constentChart?.availableAmount ?? '0,00'}
          </Text>

          {invertValues && (
            <>
              <Text
                style={[
                  FontStyles.Body_2,
                  FontStyles.MutedColor,
                  FontStyles.Center
                ]}>
                Cupo utilizado
              </Text>
              <Text style={[FontStyles.Body_2, FontStyles.Center]}>
                ${constentChart?.amountSpent ?? '0,00'}
              </Text>
            </>
          )}

          <Text
            style={[
              FontStyles.Body_2,
              FontStyles.MutedColor,
              FontStyles.Center
            ]}>
            Cupo autorizado
          </Text>
          <Text style={[FontStyles.Body_2, FontStyles.Center]}>
            ${constentChart?.totalAmount ?? '0,00'}
          </Text>
        </View>
      </View>
    </View>
  );
};
