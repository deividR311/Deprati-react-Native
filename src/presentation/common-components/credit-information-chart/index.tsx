export * from './credit-information-chart';
export * from './donut-chart';
export * from './interface';
