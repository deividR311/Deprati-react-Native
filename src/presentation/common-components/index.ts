import BannerHome from './banner-home';
import CategoryHome from './category-home';
import StoriesHome from './stories-home';
import CarouselWelcome from '../screens/Welcome/components/CarouselWelcome';
import CategoryMain from '../screens/CategoryPage/components/CategoryMain';
import CategoryFeature from '../screens/CategoryPage/components/CategoryFeature';
import BannerCarousel from '../screens/CategoryPage/components/BannerCarousel';
import RecommendedProducts from './recommended-home/RecommendProducts';
import SimpleBanner from './banner-home/SimpleBanner';
import CMSParagraph from './CMSParagraph/CMSParagraph';
import CMSLinkButton from './CMSLinkButton/CMSLinkButton';
import DepratiBrandsRotatingImages from './DepratiBrandsRotatingImages/DepratiBrandsRotatingImages';
import BannerComponent from './banner-home/BannerComponent';
import DepratiDoubleResponsiveBanner from './DepratiDoubleResponsiveBanner/DepratiDoubleResponsiveBannerComponent';
import DepratiHomeRotatingImagesComponent from './DepratiHomeRotatingImagesComponent/DepratiHomeRotatingImagesComponent';
import { DepratiSimpleResponsiveBannerComponent } from '../screens/dashboard/PDP/components/DepratiSimpleResponsiveBannerComponent';
import ProductCarouselComponent from './productCarouselComponent';
import DepratiCloudflareVideoStreamBannerComponent from './DepratiCloudflareVideoStreamBannerComponent/DepratiCloudflareVideoStreamBannerComponent';

interface ComponentType {
  [key: string]: React.ReactNode;
}

// uid : Component
export const Components: ComponentType = {
  //Home component
  DepratiStoryContainerComponent: StoriesHome,
  SimpleBannerComponent: SimpleBanner,
  DepratiCategoryListComponent: CategoryHome,
  DepratiEmarsysProductCarouselComponent: RecommendedProducts,
  DepratiHomeRotatingImagesComponent: DepratiHomeRotatingImagesComponent,

  //Welcome Component
  DepratiWelcomeRotatingImagesComponent: CarouselWelcome,
  RotatingImagesComponent: CarouselWelcome,

  //Category Component
  DepratiCategoryFeatureCarouselComponent: CategoryMain,
  CategoryFeatureComponent: CategoryFeature,
  DepratiDoubleResponsiveBannerComponent: DepratiDoubleResponsiveBanner, //BannerCarousel
  DepratiSimpleResponsiveBannerComponent,
  CMSParagraphComponent: CMSParagraph,
  CMSLinkComponent: CMSLinkButton,
  DepratiBrandsRotatingImagesComponent: DepratiBrandsRotatingImages,
  BannerComponent: BannerComponent,
  DepratiCloudflareVideoStreamBannerComponent,
  SimpleResponsiveBannerComponent: SimpleBanner,
  //Corner
  ProductCarouselComponent
};
