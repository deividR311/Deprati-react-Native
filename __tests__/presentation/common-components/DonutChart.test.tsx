import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../__mocks__/navigation-wrapper';
import { DonutChart } from '../../../src/presentation/common-components/credit-information-chart';

let screenTest: any;
describe('DonutChart', () => {
  it('renders correctly DonutChart in Snapshot', () => {
    screenTest = render(<NavigationWrapper screen={() => <DonutChart />} />);
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
