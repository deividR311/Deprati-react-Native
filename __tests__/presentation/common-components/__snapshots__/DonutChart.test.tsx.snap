// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart renders correctly DonutChart in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={200}
            bbWidth={200}
            fill="gray"
            focusable={false}
            height={200}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 200,
                  "width": 200,
                },
              ]
            }
            vbHeight={228}
            vbWidth={228}
            width={200}
          >
            <RNSVGGroup
              fill={4286611584}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                matrix={
                  Array [
                    1,
                    0,
                    -0,
                    1,
                    0,
                    0,
                  ]
                }
              >
                <RNSVGCircle
                  cx="50%"
                  cy="50%"
                  fill={0}
                  matrix={
                    Array [
                      -0.8660254037844386,
                      0.5000000000000001,
                      -0.5000000000000001,
                      -0.8660254037844386,
                      269.726896031426,
                      155.72689603142598,
                    ]
                  }
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                      "strokeDashoffset",
                      "strokeLinejoin",
                    ]
                  }
                  r={100}
                  stroke={4293322470}
                  strokeDasharray={
                    Array [
                      628.3185307179587,
                      628.3185307179587,
                    ]
                  }
                  strokeDashoffset={209.43951023931956}
                  strokeLinejoin={1}
                  strokeWidth={14}
                />
                <RNSVGCircle
                  cx="50%"
                  cy="50%"
                  fill={0}
                  matrix={
                    Array [
                      -0.8660254037844386,
                      0.5000000000000001,
                      -0.5000000000000001,
                      -0.8660254037844386,
                      269.726896031426,
                      155.72689603142598,
                    ]
                  }
                  propList={
                    Array [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                      "strokeDashoffset",
                      "strokeLinejoin",
                    ]
                  }
                  r={100}
                  stroke={4293073452}
                  strokeDasharray={
                    Array [
                      628.3185307179587,
                      628.3185307179587,
                    ]
                  }
                  strokeDashoffset={209.43951023931956}
                  strokeLinejoin={1}
                  strokeWidth={14}
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
